easybuild/easyblocks/generic/systemmpi.py

```diff
@@ -0,0 +1,160 @@
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for using (already installed/existing) system MPI instead of a full install via EasyBuild.
+
+@author Alan O'Cais (Juelich Supercomputing Centre)
+"""
+import os
+import re
+
+from easybuild.easyblocks.generic.bundle import Bundle
+from easybuild.tools.filetools import read_file, which
+from easybuild.tools.run import run_cmd
+from easybuild.framework.easyconfig.easyconfig import ActiveMNS
+from easybuild.tools.build_log import EasyBuildError
+
+
+class SystemMPI(Bundle):
+    """
+    Support for generating a module file for the system compiler with specified name.
+
+    The compiler is expected to be available in $PATH, required libraries are assumed to be readily available.
+
+    Specifying 'system' as a version leads to using the derived compiler version in the generated module;
+    if an actual version is specified, it is checked against the derived version of the system compiler that was found.
+    """
+
+    def extract_ompi_version(self, txt):
+        """Extract MPI version from provided string."""
+        # look for 3-4 digit version number, surrounded by spaces
+        # examples:
+        # gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-11)
+        # Intel(R) C Intel(R) 64 Compiler XE for applications running on Intel(R) 64, Version 15.0.1.133 Build 20141023
+        version_regex = re.compile(r'\s(v[0-9]+(?:\.[0-9]+){1,3})\s', re.M)
+        res = version_regex.search(txt)
+        if res:
+            self.mpi_version = res.group(1)[1:]
+            self.log.debug("Extracted MPI version '%s' from: %s", self.mpi_version, txt)
+        else:
+            raise EasyBuildError("Failed to extract OpenMPI version using regex pattern '%s' from: %s",
+                                 version_regex.pattern, txt)
+
+    def __init__(self, *args, **kwargs):
+        """Extra initialization: determine system compiler version and prefix."""
+        super(SystemMPI, self).__init__(*args, **kwargs)
+
+        # Determine MPI path (real path, with resolved symlinks)
+        mpi_name = self.cfg['name'].lower()
+        mpi_c_compiler = 'mpicc'
+        path_to_mpi_c_compiler = which(mpi_c_compiler)
+        if path_to_mpi_c_compiler:
+            path_to_mpi_c_compiler = os.path.realpath(path_to_mpi_c_compiler)
+            self.log.info("Found path to MPI implementation '%s' mpicc compiler (with symlinks resolved): %s",
+                          mpi_name, path_to_mpi_c_compiler)
+        else:
+            raise EasyBuildError("%s not found in $PATH", mpi_c_compiler)
+
+        # Determine compiler version and installation prefix
+        if mpi_name == 'openmpi':
+            out, _ = run_cmd("ompi_info --version", simple=False)
+            self.extract_ompi_version(out)
+
+            # extract the installation prefix
+            self.mpi_prefix, _ = run_cmd("ompi_info --path prefix|awk '{print $2}'", simple=False)
```

Comment:
@ocaisa You mean something like `res = re.search('^\s+Open MPI: (.*)$', output_of_ompi_info)` and then `res.group(1)`?